***Case Study on Testing of Hypothesis***

A company started to invest in digital marketing as a new way of their product
promotions. For that they collected data and decided to carry out a study on it


*   The company wishes to clarify whether there is any increase in sales after
stepping into digital marketing.
*  The company needs to check whether there is any dependency between
thefeatures “Region” and “Manager”.



In [62]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

In [63]:
from google.colab import files
files.upload()

Saving Sales_add.csv to Sales_add (1).csv


{'Sales_add.csv': b'Month,Region,Manager,Sales_before_digital_add(in $),Sales_After_digital_add(in $)\r\nMonth-1,Region - A,Manager - A,132921,270390\r\nMonth-2,Region - A,Manager - C,149559,223334\r\nMonth-3,Region - B,Manager - A,146278,244243\r\nMonth-4,Region - B,Manager - B,152167,231808\r\nMonth-5,Region - C,Manager - B,159525,258402\r\nMonth-6,Region - A,Manager - B,137163,256948\r\nMonth-7,Region - C,Manager - C,130625,222106\r\nMonth-8,Region - A,Manager - A,131140,230637\r\nMonth-9,Region - B,Manager - C,171259,226261\r\nMonth-10,Region - C,Manager - B,141956,193735\r\nMonth-11,Region - B,Manager - C,159339,203175\r\nMonth-12,Region - A,Manager - A,178939,276279\r\nMonth-13,Region - B,Manager - A,145062,205180\r\nMonth-14,Region - A,Manager - A,151514,253131\r\nMonth-15,Region - C,Manager - A,147463,229336\r\nMonth-16,Region - A,Manager - B,177195,187305\r\nMonth-17,Region - A,Manager - C,140862,234908\r\nMonth-18,Region - C,Manager - B,167996,191517\r\nMonth-19,Region - B,Ma

In [64]:
data = pd.read_csv('Sales_add.csv')

In [65]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [66]:
data.size

110

In [67]:
data.index

RangeIndex(start=0, stop=22, step=1)

In [68]:
data.nunique

<bound method DataFrame.nunique of        Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Re

In [69]:
data.shape

(22, 5)

In [70]:
data.isna()

Month  Region  Manager  Sales_before_digital_add(in $)  \
0   False   False    False                           False   
1   False   False    False                           False   
2   False   False    False                           False   
3   False   False    False                           False   
4   False   False    False                           False   
5   False   False    False                           False   
6   False   False    False                           False   
7   False   False    False                           False   
8   False   False    False                           False   
9   False   False    False                           False   
10  False   False    False                           False   
11  False   False    False                           False   
12  False   False    False                           False   
13  False   False    False                           False   
14  False   False    False                           False   
15  False   False    False                           False   
16  False   False    False                           False   
17  False   False    False                           False   
18  False   False    False                           False   
19  False   False    False                           False   
20  False   False    False                           False   
21  False   False    False                           False   

    Sales_After_digital_add(in $)  
0                           False  
1                           False  
2                           False  
3                           False  
4                           False  
5                           False  
6                           False  
7                           False  
8                           False  
9                           False  
10                          False  
11                          False  
12                          False  
13                          False  
14                          False  
15                          False  
16                          False  
17                          False  
18                          False  
19                          False  
20                          False  
21                          False

In [71]:
data.columns

Index(['Month', 'Region', 'Manager', 'Sales_before_digital_add(in $)',
       'Sales_After_digital_add(in $)'],
      dtype='object')

**Descriptive Analysis**

In [72]:
data.mode

<bound method DataFrame.mode of        Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Regio

In [73]:
data['Sales_before_digital_add(in $)'].mean()

149239.95454545456

In [74]:
data['Sales_After_digital_add(in $)'].mean()

231123.72727272726

In [75]:
data['Sales_before_digital_add(in $)'].median()

147444.0

In [76]:
data['Sales_After_digital_add(in $)'].median()

229986.5

In [77]:
data['Sales_before_digital_add(in $)'].var()

220345610.2359307

In [78]:
data['Sales_After_digital_add(in $)'].var()

653148853.7316017

In [79]:
data['Sales_before_digital_add(in $)'].std()

14844.04292084642

In [80]:
data['Sales_After_digital_add(in $)'].std()

25556.777060725042

In [81]:
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

**Inference**

From this it is clear that the mean sales value increased from **149239.954545**	to **231123.727273** after the introduction of Digital Marketing.

**1.The company wishes to clarify whether there is any increase in sales after stepping into digital marketing** 

In this population total number of rows and columns are 22 and 5 respectively. 

Here sample size is less than 30. So use One tailed Paired Samples T-Test.

The Hypothesis are the following:

H0:The sales before and after the introduction of digital marketing are same.

Ha:The sales after stepping into digital marketing is more.

Alpha:0.05

In [82]:
from scipy.stats import ttest_ind

In [83]:
Sales_before = data['Sales_before_digital_add(in $)']

In [84]:
Sales_after = data['Sales_After_digital_add(in $)']

In [85]:
t_test,p_value = ttest_ind(Sales_after,Sales_before)
print('t-value is:',t_test)
print('p-value is:',p_value)
if p_value<0.05:
  print('Null hypothesis is rejected')
else:
  print('null hypothesis is accepted')

t-value is: 12.995084451110877
p-value is: 2.614368006904645e-16
Null hypothesis is rejected


**Inference**

Here the p-value <0.05. Hence the null hypothesis is rejected.

Hence alternate hypothesis holds true.ie the sales after stepping into digital marketing is more.

**2.The company needs to check whether there is any dependency between the features “Region” and “Manager”.**

**Chi Square test**

The Hypothesis are the following:

H0:The 2 features “Region” and “Manager” which are to be compared are independent.

Ha:The 2 features “Region” and “Manager” which are to be compared are dependent.

Alpha:0.05

In [86]:
from scipy.stats import chi2_contingency

In [87]:
crosstab = pd.crosstab(data['Region'], data['Manager'], normalize='index')
crosstab


Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A     0.400000     0.300000     0.300000
Region - B     0.571429     0.142857     0.285714
Region - C     0.200000     0.600000     0.200000

In [88]:
c, p, dof, expected = chi2_contingency(crosstab)
print('chi square statistic:',c)
print('p-value:',p)
print('degrees of freedom:',dof)
if p<0.05:
  print('Null hypothesis is rejected')
else:
  print('null hypothesis is accepted')

chi square statistic: 0.5097129666190809
p-value: 0.9725485584250712
degrees of freedom: 4
null hypothesis is accepted


**Inference**

Here the p-value is greater than 0.05. Hence null hypothesis is accepted. Which means 'Region' and 'Manager' are indepedent.That is no relationship between the Region and Manager.Also chi square statistic less than critical value,so we cannot reject null hypothesis.

**Result**



1.  The sales after stepping into digital marketing is more.
2.  There is no relationship between the Region and Manager

